packages/stark-ui/src/modules/transform-input-directive/directives/transform-input.directive.ts

```diff
@@ -0,0 +1,141 @@
+import { Directive, ElementRef, forwardRef, Input, OnChanges, Renderer2, SimpleChanges } from "@angular/core";
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from "@angular/forms";
+
+export const UPPERCASE: "uppercase" = "uppercase";
+export const LOWERCASE: "lowercase" = "lowercase";
+export type StarkInputTransformationType = "uppercase" | "lowercase" | ((value: any) => any);
+
+/**
+ * Provider for StarkTransformInputDirective
+ */
+export const STARK_TRANSFORM_INPUT_PROVIDER: any = {
+	provide: NG_VALUE_ACCESSOR,
+	// tslint:disable-next-line:no-forward-ref
+	useExisting: forwardRef(() => StarkTransformInputDirective),
+	multi: true
+};
+
+/**
+ * Directive to transform the value of an input / textarea before it is passed on to the ngControl (ngModel, formControl).
```

Comment:
Can you also mention the fact that this directive does not interfere with the programmatic changes to the ngModel/FormControl?